lib/src/native/realm_core.dart

```diff
@@ -148,40 +147,173 @@ class _RealmCore {
+        try {
+          _realmLib.realm_sync_config_set_session_stop_policy(syncConfigPtr, config.sessionStopPolicy.index);
+          _realmLib.realm_config_set_sync_config(configPtr, syncConfigPtr);
+        } finally {
+          _realmLib.realm_release(syncConfigPtr.cast());
+        }
       }
 
       return configHandle;
     });
   }
 
+  SubscriptionSetHandle getSubscriptions(Realm realm) {
+    return SubscriptionSetHandle._(_realmLib.invokeGetPointer(() => _realmLib.realm_sync_get_active_subscription_set(realm.handle._pointer)));
+  }
+
+  void refreshSubscriptions(SubscriptionSet subscriptions) {
+    _realmLib.invokeGetBool(() => _realmLib.realm_sync_subscription_set_refresh(subscriptions.handle._pointer));
+  }
+
+  int getSubscriptionSetSize(SubscriptionSet subscriptions) {
+    return _realmLib.realm_sync_subscription_set_size(subscriptions.handle._pointer);
+  }
+
+  SubscriptionHandle subscriptionAt(SubscriptionSet subscriptions, int index) {
+    return SubscriptionHandle._(_realmLib.invokeGetPointer(() => _realmLib.realm_sync_subscription_at(
+          subscriptions.handle._pointer,
+          index,
+        )));
+  }
+
+  SubscriptionHandle? findSubscriptionByName(SubscriptionSet subscriptions, String name) {
+    return using((arena) {
+      return _realmLib
+          .realm_sync_find_subscription_by_name(
+            subscriptions.handle._pointer,
+            name.toUtf8Ptr(arena),
+          )
+          .convert(SubscriptionHandle._);
+    });
+  }
+
+  SubscriptionHandle? findSubscriptionByQuery(SubscriptionSet subscriptions, RealmResults query) {
+    return _realmLib
+        .realm_sync_find_subscription_by_query(
+          subscriptions.handle._pointer,
+          query.queryHandle._pointer,
+        )
+        .convert(SubscriptionHandle._);
+  }
+
+  static void _stateChangeCallback(Pointer<Void> userdata, int state) {
+    final completer = userdata.toObject<Completer<int>>(isPersistent: true);
+    if (completer == null) {
+      return;
+    }
+
+    // TODO: What about errors?!
+
+    completer.complete(state);
+  }
+
+  Future<int> waitForSubscriptionSetStateChange(SubscriptionSet subscriptions, SubscriptionSetState notifyWhen) {
+    final completer = Completer<int>();
+    _realmLib.realm_dart_sync_on_subscription_set_state_change_async(
+      subscriptions.handle._pointer,
+      notifyWhen.index,
+      Pointer.fromFunction(_stateChangeCallback),
+      completer.toPersistentHandle(),
+      _deletePersistentHandleFuncPtr,
+      subscriptions.realm.scheduler.handle._pointer,
+    );
+    return completer.future;
+  }
+
+  int subscriptionSetVersion(SubscriptionSet subscriptions) {
+    return _realmLib.realm_sync_subscription_set_version(subscriptions.handle._pointer);
+  }
+
+  int subscriptionSetState(SubscriptionSet subscriptions) {
+    refreshSubscriptionSet(subscriptions);
```

Comment:
I have removed it from here, but added it to `SubscriptionSetState._waitForStateChange`. The problem is that it would be odd to await a state, then read the state and get a different result.